meshdb/meshdb.go

```diff
@@ -240,6 +240,20 @@ func (m *MeshDB) FindLatestMiniHeader() (*miniheader.MiniHeader, error) {
 	return miniHeaders[0], nil
 }
 
+// ClearAllMiniHeaders removes all stored MiniHeaders from the database.
+func (m *MeshDB) ClearAllMiniHeaders() error {
+	var storedHeaders []*miniheader.MiniHeader
+	if err := m.MiniHeaders.FindAll(&storedHeaders); err != nil {
```

Comment:
This is really inefficient and could be improved with some changes to the `db` package. I'll make a separate issue for that.

For now, can you use a transaction here? Would ensure that:

1. We have consistency between the calls to `FindAll` and `Delete` and the state doesn't change in the middle of this function.
2. We perform all the deletes at once in a single atomic batch.